lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -390,6 +390,10 @@ probe_cb(void *cb_ctx, const struct spdk_nvme_probe_info *probe_info,
 		return false;
 	}
 
+	if (ctx->controllers_remaining > 0) {
```

Comment:
I'm not really sure what the behavior should be when NumControllers is set and explicit BDF addresses are also provided.  NumControllers is really mainly meant for use with the default setup where no BDFs are specified.  I think your interpretation is probably reasonable, or we could change the configuration file parsing code to disallow this case (e.g. only allow BDF or NumControllers, not both).